common/resource/fx.go

```diff
@@ -107,6 +107,7 @@ var Module = fx.Options(
 		fx.ResultTags(`group:"deadlockDetectorRoots"`),
 	)),
 	fx.Provide(serialization.NewSerializer),
+	fx.Provide(persistence.NewHistoryBranchUtil),
```

Comment:
Just what to confirm that we have an (easy)  way to inject a different impl if we want to.